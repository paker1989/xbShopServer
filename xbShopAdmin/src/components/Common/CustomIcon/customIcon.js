import React from 'react';
import { Icon } from 'antd';

const CustomIcon = ({ name, ...nativeProps }) => {
    /* eslint-disable */
    // const svg = require(`../../../static/image/icon/${name}.svg`);
    /* eslint-enable */
    // console.log(svg);
    const component = () => <img alt={name} src={`/static/image/icon/${name}.svg`} />;
    return <Icon component={component} {...nativeProps} />;
};

export default CustomIcon;
