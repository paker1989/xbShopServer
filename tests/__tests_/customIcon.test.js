import React from 'react';
import renderer from 'react-test-renderer';

import CustomIcon from '../../xbShopAdmin/src/components/Common/CustomIcon/customIcon';

it('should match customIcon snapshot', () => {
    const component = renderer.create(<CustomIcon />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
