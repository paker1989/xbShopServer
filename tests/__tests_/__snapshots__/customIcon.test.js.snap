// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match customIcon snapshot 1`] = `
<i
  className="anticon"
>
  <img
    src="/static/image/icon/undefined.svg"
  />
</i>
`;
